Objects/typeobject.c

```diff
@@ -404,7 +405,13 @@ set_tp_mro(PyTypeObject *self, PyObject *mro)
         /* Other checks are done via set_tp_bases. */
         _Py_SetImmortal(mro);
     }
-    self->tp_mro = mro;
+#ifdef Py_GIL_DISABLED
+    if (self->tp_mro != NULL) {
+        // Allow concurrent reads from PyType_IsSubtype
+        _PyObject_GC_SET_SHARED(self->tp_mro);
```

Comment:
I think this should be `_PyObject_GC_SET_SHARED_INLINE()`.

The naming is pretty confusing to me. I think we should try to come up with better names than `_PyGC_BITS_SHARED` and `_PyGC_BITS_SHARED_INLINE`.